// Copyright 2015 gRPC authors.
//
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
//     http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.

syntax = "proto3";

option java_multiple_files = true;
option java_package = "io.grpc.examples.helloworld";
option java_outer_classname = "HelloWorldProto";
option objc_class_prefix = "HLW";

package helloworld;

// The greeting service definition.
service Getter {
  // Sends a greeting
  rpc GetIMDBData (Request) returns (Result) {}
}

// The request message containing the user's name.
message Request {
  int32 rowOffset = 1; 
}

// The response message containing the greetings
message Result {
  message Row {
    string tconst = 1;
    string titleType = 2;
    string primaryTitle = 3;
    string originalTitle = 4;
    bool isAdult = 5;
    string startYear = 6;
    string endYear = 7;
    string runtimeMinutes = 8;
    string genres = 9;
  }
  repeated Row results = 1;
}
